base_layer/core/tests/support/simple_block_chain.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019 The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use chrono::Duration;
+use merklemountainrange::mmr::*;
+use rand::OsRng;
+use tari_core::{block::*, blockheader::*, pow::*, transaction::*, types::*};
+use tari_crypto::{keys::SecretKey, ristretto::*};
+use tari_utilities::hash::Hashable;
+
+/// This is used to represent a block chain in memory for testing purposes
+pub struct SimpleBlockChain {
+    blocks: Vec<Block>,
+    headers: MerkleMountainRange<BlockHeader, SignatureHash>,
+    utxos: MerkleMountainRange<TransactionInput, SignatureHash>,
+    kernels: MerkleMountainRange<TransactionKernel, SignatureHash>,
+}
+impl SimpleBlockChain {
+    /// This will create a new test block_chain
+    pub fn new(amount: u32) -> SimpleBlockChain {
```

Comment:
nit: rename amount to something like num_blocks so that its clear what it means.

Reply:
Renamed for the next PR.